### PR TITLE
Add DAG Labels

### LINKED ISSUE #216: use human-readable labels for `redcap_data_access_group`?
# Feature Request Description
I'm analyzing a REDCap database with Data Access Groups, and the `redcap_data_access_group` variable in the tibble that comes out of `read_redcap(raw_or_label = "label") |> extract_tibble()` has shorthand labels, like `boston_childrens_h`:

Image

In contrast, the REDCap browser interface shows nicely formatted labels, like:

Image

I'm trying to get those human-readable labels into my tibbles (without recoding `redcap_data_access_group` manually).

# Proposed Solution

Ideally, the default behavior of `read_redcap(raw_or_label = "label")` would output the human-readable labels as the factor levels of `redcap_data_access_group`.

**Describe alternatives you've considered**

I'm guessing the human-readable labels are already stored somewhere in the output of `read_redcap()`, but I'm not sure where, and it would be great if there were an easy way to get them automatically loaded into `redcap_data_access_group`.
Thanks for any solutions you can suggest, and apologies if there's already a documented answer that I've missed! I checked 
https://chop-cgtinformatics.github.io/REDCapTidieR/articles/diving_deeper.html?q=access%20group#data-access-groups, in the example it looks like the shorthands are being output there as well:

Image

## Additional Context

Might be related to #14 or #186?

### Checklist

- [ ] The issue is atomic
- [ ] The issue description is documented
- [ ] The issue title describes the problem succinctly
- [ ] Developers are assigned to the issue
- [ ] Labels are assigned to the issue

### PR DESCRIPTION
**Pull request by rsh52**
# Description
This PR seeks to incorporate DAG labels that have been missing.  REDCapTidieR retrieves raw data by default, then merges on labels from the metadata. However, some columns like `redcap_data_access_group` are not in the metadata and so there is nothing by default to reference.

# Proposed Changes 
List changes below in bullet format:

- Introduce new utility function `update_dag_cols()` which only gets called when:
   - `export_data_access_groups == TRUE` AND
   -  `raw_or_label == "label"` 
      - Unsure if this should also be when chosen as "haven" @ezraporter 
- New test added to test suite
- `redcap_dag_read()` added to NAMESPACE imports from REDCapR

### Issue Addressed
**Closes #216**

### PR Checklist
Before submitting this PR, please check and verify below that the submission meets the below criteria:

- [x] New/revised functions have associated tests
- [NA] New/revised functions that update downstream outputs have associated static testing files (`.RDS`) updated under `inst/testdata/create_test_data.R`
- [x] New/revised functions use appropriate naming conventions
- [x] New/revised functions don't repeat code
- [x] Code changes are less than **250** lines total
- [x] Issues linked to the PR using [GitHub's list of keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] The appropriate reviewer is assigned to the PR
- [x] The appropriate developers are assigned to the PR
- [x] Pre-release package version incremented using `usethis::use_version()`

# Code Review
This section to be used by the reviewer and developers during Code Review after PR submission

### Code Review Checklist

- [ ] I checked that new files follow naming conventions and are in the right place
- [ ] I checked that documentation is complete, clear, and without typos
- [ ] I added/edited comments to explain "why" not "how"
- [ ] I checked that all new variable and function names follow naming conventions
- [ ] I checked that new tests have been written for key business logic and/or bugs that this PR fixes
- [ ] I checked that new tests address important edge cases

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1209262635222723

### CODE REVIEW

**Review by ezraporter — CHANGES_REQUESTED** [review]
See my comments on how to handle `raw_or_label == "haven"` and what happens when `export_data_access_groups = NULL`.

We also need to think about how this interacts with the workflow described [here](https://chop-cgtinformatics.github.io/REDCapTidieR/articles/labelled.html#managing-variable-labels) where the user does:

```r
read_redcap(
  redcap_uri,
  token
) |>
  make_labelled()
```

Let me think about that a bit more

**Comment by ezraporter** [comment]
Also found a couple missing variable labels while investigating this so I added them in this https://github.com/CHOP-CGTInformatics/REDCapTidieR/pull/218

File: R/read_redcap.R [hunk 1]
```diff
@@ -572,3 +582,53 @@ get_repeat_event_types <- function(data) {
...
+
+  # Test dataset resembling REDCapTidieR redcap_dag_read output
+  if (test_data) {
+    dag_data <- tibble::tribble(
+      ~"data_access_group_name", ~"unique_group_name", ~"data_access_group_id",
+      "DAG1", "dag1", 28130,
+      "DAG2", "dag2", 28131,
+      "DAG3", "dag3", 28132
+    )
+  } else {
```
[the remark this answers is not in the document]
Comment (rsh52) [reply R/read_redcap.R:617]: I'm ambivalent, figured this would come up but sided on making a testable way to prove this does what we want. I can move the API call outside of this function.

File: R/read_redcap.R [hunk 2]
```diff
@@ -199,6 +198,17 @@ read_redcap <- function(redcap_uri,
...
+      # If DAGs requested in label format, trigger an API call and update column
+      # data for redcap_data_access_group
+      if (raw_or_label == "label") {
+        db_data <- update_dag_cols(
+          data = db_data,
+          redcap_uri = redcap_uri,
+          token = token,
+          suppress_redcapr_messages = !suppress_redcapr_messages
+        )
+      }
```
[the remark this answers is not in the document]
Comment (rsh52) [reply R/read_redcap.R:211]: Ah good catch, you're right. I forgot that we attempt to auto-detect versus REDCapR where the user has to be explicit.

**Comment by rsh52** [comment]
Think I hit everything but lmk your thoughts!

File: R/read_redcap.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (ezraporter) [reply R/read_redcap.R]: I don't think the fix you put in does it because `update_dag_cols()` won't run if `xport_data_access_groups = NULL` still:

```r
devtools::load_all()

data <- read_redcap(
  Sys.getenv("REDCAP_URI"),
  Sys.getenv("REDCAPTIDIER_DAG_API")
)

data |>
  extract_tibble("non_repeat_form_1")
```

Is there an issue with using just `if(export_data_access_groups)` here:

https://github.com/CHOP-CGTInformatics/REDCapTidieR/blob/93ebfd533ccdb33e391795f81164f11ed2d95bca/R/read_redcap.R#L195

And this can just check `raw_or_label != "raw"`:

https://github.com/CHOP-CGTInformatics/REDCapTidieR/blob/93ebfd533ccdb33e391795f81164f11ed2d95bca/R/read_redcap.R#L205

**Review by ezraporter — CHANGES_REQUESTED** [review]
I don't think this quite fixed the issue with `export_data_access_groups = NULL`. See my example and suggestion there.

**Comment by rsh52** [comment]
> I don't think this quite fixed the issue with `export_data_access_groups = NULL`. See my example and suggestion there.

Ok! Trying a different approach because I don't really like having to figure out the `export_*`/`export_*_original` bit (I was keeping it inside the `!is.null()` check because if `NULL` the check for T/F would fail). Instead, looking for the existence of `redcap_data_access_group` in the names of the database output and updating that column (if it exists) based on designation of raw/label/haven.

I could be convinced this is fragile, but at the moment I can't think of a real reason why. If we agree it's ok, then this means that if `export_data_access_groups` remains `NULL` we will automatically update the `redcap_data_access_group` column to label/haven if specified. I think this is actually what we want and required an update to a couple of tests.

**Review by ezraporter — APPROVED** [review]
New approach makes sense to me!
